src/halo_comm_x.f90

```diff
@@ -0,0 +1,153 @@
+!=======================================================================
+! This is part of the 2DECOMP&FFT library
+!
+! 2DECOMP&FFT is a software framework for general-purpose 2D (pencil)
+! decomposition. It also implements a highly scalable distributed
+! three-dimensional Fast Fourier Transform (FFT).
+!
+! Copyright (C) 2009-2021 Ning Li, the Numerical Algorithms Group (NAG)
+!
+!=======================================================================
+
+! This file contain common code to be included by subroutines
+! 'exchange_halo_x_...' in halo.f90
+
+    if (present(opt_decomp)) then
+      decomp = opt_decomp
+    else
+      decomp = decomp_main
```

Comment:
See transpose routines to avoid a copy of the decomp object

Reply:
Done